### ISSUE TITLE
Diverged metrics in PatchCore: dropping from 0.99 to 0.44 and 0.03 is rather critical?

### ISSUE THREAD

**Issue by samet-akcay** [openvinotoolkit/anomalib#74]
diverged metrics: dropping from 0.99 to 0.44 and 0.03 is rather critical?
log images: nice :) 
Also padim dropped in performance, but not as crazy.
Here's a patchcore result of "good" parts: 
![008](https://user-images.githubusercontent.com/30144796/148973398-5e317afa-b682-4c44-9131-448fb19a64f2.png)

this is padim:
DATALOADER:0 TEST RESULTS
{'image_AUROC': 0.7589669823646545,
 'image_F1': 0.8787878751754761,
 'pixel_AUROC': 0.9781586527824402,
 'pixel_F1': 0.22379672527313232}
![008padim](https://user-images.githubusercontent.com/30144796/148974485-86b96972-a13f-4de2-b977-2036ea9e6f5e.png)

_Originally posted by @sequoiagrove in https://github.com/openvinotoolkit/anomalib/issues/67#issuecomment-1010091154_

**Comment by samet-akcay** [comment]
@sequoiagrove, @blakshma worked on the patchcore results, where he managed to improve the performance to the following:

```
DATALOADER:0 TEST RESULTS
{'image_AUROC': 0.9524492025375366,
 'image_F1': 0.9551020860671997,
 'pixel_AUROC': 0.9894225597381592,
 'pixel_F1': 0.3562552034854889}
```

To reproduce the numbers you could use [this branch](https://github.com/openvinotoolkit/anomalib/tree/fix/barath/patchcore), to be merged to development soon after [this PR](https://github.com/openvinotoolkit/anomalib/pull/73).

Here are the qualitative results for the ones you shared above. (screw/test/good/009/png)
![image](https://user-images.githubusercontent.com/17574420/149715268-ff32518b-87d1-44cb-a900-e832138f831d.png)

**Comment by sequoiagrove** [comment]
I get those numbers too now. but I wonder of the computation of the numbers is also wrong, because when I look through the results images, it only detects about 1/4 to 1/2 of the defects in the different categories. The pixel F1 = 0.35 seems to decribe best the actual performance. but I know that it is low due to correctly detected defects dont have to overlap perfectly to still be a good result.

**Comment by sequoiagrove** [comment]
pixel auroc of 0.97 sound like it is good performance, but looking at the masks it is really not useful in a real system, and the auc is a bad metric for quantifying performance.
So I guess I should always look at pixel F1. Think thats about the same as dice, right? I use dice in segmentation tasks.
Too bad this list shows only the unrealistically positive auroc numbers:
https://paperswithcode.com/sota/anomaly-detection-on-mvtec-ad

**Comment by samet-akcay** [comment]
yeah, AUC is widely used in academia, but usually not a good metric for industrial applications since it could be misleading. Finding the best threshold from the AUC is not so easy even though we implemented adaptive thresholding mechanism. This is the reason why we also added f1 score to our evaluations. 

Regarding always looking at the pixel f1 score for evaluation, there is room there for improvement. We haven't optimise our heatmaps from which the predicted masks are generated. Once we do, I agree, pixel f1 would become the standard metric to evaluate the performance.
